sys/include/net/gnrc/netif/internal.h

```diff
@@ -346,14 +346,33 @@ static inline bool gnrc_netif_is_rtr_adv(const gnrc_netif_t *netif)
  * @return  true, if the interface represents a 6LN
  * @return  false, if the interface does not represent a 6LN
  */
-#if ((GNRC_NETIF_NUMOF > 1) && defined(MODULE_GNRC_SIXLOWPAN)) || \
-    defined(MODULE_GNRC_SIXLOENC) || defined(DOXYGEN)
-bool gnrc_netif_is_6lo(const gnrc_netif_t *netif);
-#elif (GNRC_NETIF_NUMOF == 1) && defined(MODULE_GNRC_SIXLOWPAN)
-#define gnrc_netif_is_6lo(netif)                (true)
-#else
-#define gnrc_netif_is_6lo(netif)                (false)
+static inline bool gnrc_netif_is_6lo(const gnrc_netif_t *netif)
+{
+    if((!gnrc_netif_single() &&
+       IS_USED(MODULE_GNRC_SIXLOWPAN)) || \
+       IS_USED(MODULE_GNRC_SIXLOENC)) {
+        switch (netif->device_type) {
+#ifdef MODULE_GNRC_SIXLOENC
+            case NETDEV_TYPE_ETHERNET:
+                return (netif->flags & GNRC_NETIF_FLAGS_6LO);
 #endif
+            case NETDEV_TYPE_IEEE802154:
+            case NETDEV_TYPE_CC110X:
+            case NETDEV_TYPE_BLE:
+            case NETDEV_TYPE_NRFMIN:
+            case NETDEV_TYPE_ESP_NOW:
+                return true;
+            default:
+                return false;
+        }
+    }
```

Comment:
Have you checked the memory impact of moving this to `static inline`? `examples/gnrc_border_router` should be the application to look at.

Reply:
yes, it increased. The last commit fixes that problem (`gnrc_netif_is_6lo` is still static inline but I added a non-static function to check if the device requires 6Lo).

Now `gnrc_border_router` has the same size with and without this PR :)